pkg/operator/ceph/cluster/controller.go

```diff
@@ -343,6 +343,8 @@ func (c *ClusterController) reconcileCephCluster(clusterObj *cephv1.CephCluster,
 		cluster = newCluster(clusterObj, c.context, ownerInfo)
 	}
 	cluster.namespacedName = c.namespacedName
+	// updating observedGeneration in cluster if it's not the first reconcile
+	cluster.observedGeneration = clusterObj.ObjectMeta.Generation
```

Comment:
Added this change, now it will work, currently, it was talking value of first reconcile value only, 
Thanks!